.github/workflows/run-integration-tests.yaml

```diff
@@ -0,0 +1,131 @@
+#     permissions:
+#       contents: 'read'
+#      with:
+#        test_url: 'https://bvdp-saturn-dev.appspot.com/'
+#        environment: 'dev'
+#        tests_to_run: 'tests/register-user.js'
+#      secrets:
+#        LYLE_CLIENT_EMAIL: ${{ secrets.LYLE_CLIENT_EMAIL }}
+#        LYLE_PRIVATE_KEY: ${{ secrets.LYLE_PRIVATE_KEY }}
+#        FIRECLOUD_ACCOUNT_CLIENT_EMAIL: ${{ secrets.FIRECLOUD_ACCOUNT_CLIENT_EMAIL }}
+#        FIRECLOUD_ACCOUNT_PRIVATE_KEY: ${{ secrets.FIRECLOUD_ACCOUNT_PRIVATE_KEY }}
+# ```
+
+on:
+  workflow_call:
+    inputs:
+
+      ##
+      ## Configurations:
+      ##
+
+      test_url:
+        required: true
+        type: string
+        description: "URL of terraui instance to test against, required"
+      environment:
+        required: false
+        type: string
+        default: dev
+        description: "environment config to use, default: dev"
+      tests_to_run:
+        required: false
+        type: string
+        description: "space delimited list of js test files to run, default: run all"
+      target_branch:
+        required: false
+        type: string
+        description: "The branch to run integration tests against"
+
+    # technically, the emails are not secret, but they're stored w/ the private key so
+    # we'll just pull them to avoid potential sync/typo/refresh issues.
+    secrets:
+      LYLE_CLIENT_EMAIL: 
+        required: true
+      LYLE_PRIVATE_KEY: 
+        required: true
+      FIRECLOUD_ACCOUNT_CLIENT_EMAIL:
+        required: true
+      FIRECLOUD_ACCOUNT_PRIVATE_KEY:
+        required: true
+
+env:
+  ENVIRONMENT: ${{ inputs.environment }}
+  TEST_URL: ${{ inputs.test_url }}
+
+jobs:
+  run-integration-tests:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: 'read'
+    steps:
+
+      ##
+      ## Handle Checkout:
+      ##
+
+      - name: "Checkout terra-ui code"
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.target-branch }}
+          repository: databiosphere/terra-ui
+          path: terraui
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - name: "Run Yarn Install"
+        shell: bash
+        run: cd terraui && yarn install
+
+      # Current (Jul 2023) test implementation requires SA Keys to be made available (lyle-utilts.js, terra-sa-utils.js)
+      - name: "assemble Service Account Keys"
+        run: |
+          echo \
+          LYLE_SA_KEY=$(jq --null-input \
+          --arg client_email "${{ secrets.LYLE_CLIENT_EMAIL }}" \
+          --arg key"${{ secrets.LYLE_PRIVATE_KEY }}" \
+          '{"client_email": $client_email, "key": $key}') >> $GITHUB_ENV
+          
+          echo \
+          TERRA_SA_KEY=$(jq --null-input \
+          --arg client_email "${{ secrets.FIRECLOUD_ACCOUNT_CLIENT_EMAIL }}" \
+          --arg key "${{ secrets.FIRECLOUD_ACCOUNT_PRIVATE_KEY }}" \
+          '{"client_email": $client_email, "key": $key}') >> $GITHUB_ENV
```

Comment:
Would it be simpler to have each SA key stored as a secret instead of reconstructing them from two secrets that have to be coordinated?

Reply:
I did this setup specifically because it pulls from current vault secrets that I imagine circleci also uses, it would require duplicating the secrets themselves and reformatting them to make it work, and then we'd end up w/ 2 secrets in vault for the same data that would have to be managed and maintained.

tl;dr: It would make the CI cleaner, but would make the vault setup much worse.

Reply:
Ah, didn't realize these were synced from Vault. I think CircleCI's secrets are probably manually copied and pasted, so if this is automated, then this is much better.

Reply:
Yeah, I'm working w/ the expectation that any repo that tries to call the test would be responsible for pulling the secrets into github via terraform (example below)

https://github.com/broadinstitute/terraform-ap-deployments/blob/df648b0383988781a5299e9066e0d0342f59a676/github/tfvars/broadinstitute-monkey.tfvars#L23